Specification

derf opened the pull request:
Add a jsonschema draft-07 compatible specification of the transport API schema. I deliberately left out the API-specific settings below the "options" block.

Apart from missing coverage information, our current data validates against the spec. Once we fix that, we can use https://github.com/marketplace/actions/validate-json to automatically validate all endpoint definitions, which should come in handy for pull requests. Note that validate-json requires an explicit list of files to validate, it does not yet support wildcard expansion or directories.

derf committed:
Add JSON Schema specification

derf committed:
Add GeoJSON Polygon schema

derf committed:
Use enum for accepted endpoint types

derhuerst commented on schema.json:
```diff
@@ -0,0 +1,93 @@
+          "$ref": "#coverage"
+        },
+        "anyCoverage": {
+          "$ref": "#coverage"
```
The ["structuring a complex schema" guide](https://json-schema.org/understanding-json-schema/structuring.html) uses `#/definitions/coverage`. Shall we use that?

derf replied:
Good point. Yes, let's use that.

derf committed:
Use #/definitions/... for referenced definitions

This is in line with the recommendations given on
<https://json-schema.org/understanding-json-schema/structuring.html>
